src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs

```diff
@@ -92,91 +88,116 @@ private void WriteClassHeader()
+                _targetStream.WriteLine($"        Private Const s_resourcesName As String = \"{_resourcesName}.SR\"");
                 _targetStream.WriteLine("");
-                if (!DebugOnly)
-                    _targetStream.WriteLine("#If Not DEBUGRESOURCES Then");
             }
         }
 
-        private void RunOnResFile()
+        private void WriteResources()
         {
-            foreach(KeyValuePair<string, string> pair in GetResources(ResxFilePath))
+            var resources = GetResources(ResxFilePath);
+
+            if (AsConstants)
             {
-                StoreValues((string)pair.Key, (string)pair.Value);
+                foreach (var resourcePair in resources)
+                {
+                    WriteResourceConstant((string)resourcePair.Key);
+                }
+            }
+            else
+            {
+                _targetStream.WriteLine(_targetLanguage == TargetLanguage.CSharp ?
+                    "#if !DEBUGRESOURCES" :
+                    "#If Not DEBUGRESOURCES Then");
+
+                foreach (var resourcePair in resources)
+                {
+                    WriteResourceProperty(resourcePair.Key, _targetLanguage == TargetLanguage.CSharp ?
+                        "null" :
+                        "Nothing");
+                }
+
+                _targetStream.WriteLine(_targetLanguage == TargetLanguage.CSharp ?
+                    "#else" :
+                    "#Else");
+
+                foreach (var resourcePair in resources)
+                {
+                    WriteResourceProperty(resourcePair.Key, CreateStringLiteral(resourcePair.Value));
+                }
+
+                _targetStream.WriteLine(_targetLanguage == TargetLanguage.CSharp ?
+                    "#endif" :
+                    "#End If");
             }
         }
 
-        private void StoreValues(string leftPart, string rightPart)
+        private string CreateStringLiteral(string original)
         {
-            int value;
-            if (_keys.TryGetValue(leftPart, out value))
+            StringBuilder stringLiteral = new StringBuilder(original.Length + 3);
+            if (_targetLanguage == TargetLanguage.CSharp)
             {
-                return;
+                stringLiteral.Append('@');
             }
-            _keys[leftPart] = 0;
-            StringBuilder sb = new StringBuilder(rightPart.Length);
-            for (var i = 0; i < rightPart.Length; i++)
+            stringLiteral.Append('\"');
+            for (var i = 0; i < original.Length; i++)
             {
                 // duplicate '"' for VB and C#
-                if (rightPart[i] == '\"' && (_targetLanguage == TargetLanguage.VB || _targetLanguage == TargetLanguage.CSharp))
+                if (original[i] == '\"' && (_targetLanguage == TargetLanguage.VB || _targetLanguage == TargetLanguage.CSharp))
```

Comment:
>_targetLanguage == TargetLanguage.VB || _targetLanguage == TargetLanguage.CSharp [](start = 44, length = 80)

I think we support either CSharp or VB, so this condition looks not needed at least for now.